include/ua_plugin_securitypolicy.h

```diff
@@ -178,32 +178,34 @@ typedef struct
     const size_t encryptingBlockSize;
 } UA_SecurityPolicySymmetricModule;
 
+/**
+ * This struct defines initialization Data that is potentially needed by every security policy.
+ * Some of these, like the revocation list for example, may be omitted (empty ByteString).
+ */
+typedef struct
+{
+    UA_ByteString localPrivateKey;
+    UA_ByteString localCertificate;
+    UA_ByteString certificateTrustList;
+    UA_ByteString certificateRevocationList;
+} UA_Endpoint_SecurityContext_RequiredInitData;
+
 struct UA_Endpoint_SecurityContext {
+    /**
+     * \brief Creates a new endpoint context with the supplied initialization data.
+     *
+     * \param securityPolicy the securityPolicy this function is invoked on.
+     * \param initData the required initialization data for the endpoint context.
+     */
     UA_StatusCode (*const newContext)(const UA_SecurityPolicy *securityPolicy,

```

Comment:
Currently both have a newContext function.
The channelContext only has functions for managing the symmetric keys and the remote certificate though.

You mean you want a UA_SecurityPolicy_none_EndpointContext_new(....) function?
This would mean we define a new function with the exact same signature and a different name for each security policy. (There might be some additional parameters, but that's what the void pointer is for currently)
Also, the client needs to create a context as well, and currently this can be easily done by just invoking securityPolicy->endpointContext.newContext(....). If we were to take that out of there we would first have to determine the correct function to call and make some kind of hardwired switch case maybe? Or am i overlooking something/thinking too difficult?